[Response Ops][Reporting] Scheduled Reports

ymao1 opened the pull request:
Resolves https://github.com/elastic/kibana/issues/216313

## Summary

This is a feature branch that contains the following commits. Each individual linked PR contains a summary and verification instructions.

* Schedule API - https://github.com/elastic/kibana/pull/219771
* Scheduled report task runner - https://github.com/elastic/kibana/pull/219770
* List and disable API - https://github.com/elastic/kibana/pull/220922
* Audit logging - https://github.com/elastic/kibana/pull/221846
* Send scheduled report emails - https://github.com/elastic/kibana/pull/220539
* Commit to check license - https://github.com/elastic/kibana/pull/221028/commits/f5f9d9daedcd18447b6a02335a53631a44413788
* Update to list API response format - https://github.com/elastic/kibana/pull/224262

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

ymao1 committed:
[Response Ops][Reporting] Scheduled Reports - Schedule API  (#219771)

Towards https://github.com/elastic/kibana/issues/216313

> [!Note]
> This PR will be merged into a feature branch -
`elastic:scheduled-reports`
> Once this is merged, I will open a draft PR from the feature branch
into `main`

## Summary

### Schedule API

- Added internal API `/internal/reporting/schedule/{exportTypeId}`

```
POST /internal/reporting/schedule/{exportTypeId}
{
  "jobParams": <same jobParams format as existing generate single report API>,
  "schedule": {
    "rrule": <modified Task Manager rRule schema from https://github.com/elastic/kibana/pull/217728>
  }
  "notification": {
    "email": {
      "to": <optional list of email addresses>,
      "cc": <optional list of email addresses>,
      "bcc": <optional list of email addresses>
    }
}
```

- Scheduling a report creates a new `scheduled_report` type saved object
in the `.kibana-alerting-cases` index.
- Uses the same role privileges as the current generate report API, so
user must have the reporting subfeature privilege for one of the
Analytics feature privileges (Discover/Dashboard/Visualize)

## Verify

1. Run ES and Kibana with examples and install the sample web logs data
set.
2. In the DEV console, call the schedule API using

```
POST kbn:/internal/reporting/schedule/printablePdfV2
{
    "schedule": {
        "rrule": {
            "freq": 3,
            "interval": 3,
            "byhour": [12],
            "byminute": [0]
        }
    },
    "jobParams": "(browserTimezone:America/New_York,layout:(dimensions:(height:2220,width:1364),id:preserve_layout),locatorParams:!((id:DASHBOARD_APP_LOCATOR,params:(dashboardId:edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b,preserveSavedFilters:!t,timeRange:(from:now-7d/d,to:now),useHash:!f,viewMode:view))),objectType:dashboard,title:'[Logs] Web Traffic',version:'9.1.0')"
}
```

3. Verify a saved object has been created using `GET
.kibana_alerting_cases/_search?q=type:scheduled_report`
4. Try different requests with different schedules and notifications. It
should respect the email domain allowlist if configured and reject
invalid rrule schedules and email addresses.

---------

Co-authored-by: Ersin Erdal <ersin.erdal@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ersin Erdal <92688503+ersin-erdal@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301104)
* Commit: a7065b22d1f54b111609beee0d0bcc1a66182b9e

### Failed CI Steps
* [FTR Configs #79](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eeaa-6bd8-4e82-b89f-74bf8562f849)
* [FTR Configs #79](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eee5-6585-4613-97fe-40824d19fec7)
* [FTR Configs #80](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eeaa-6bd9-488e-85bd-515004e25692)
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eeaa-6bef-402e-b741-7eac6453d069)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eeaa-6bef-402e-b741-7eac6453d069) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301104/jobs/0196eeaa-6bef-402e-b741-7eac6453d069/artifacts/0196eecc-8cf5-486c-b253-97a69d25be15) FTR Configs #<span></span>113 / Cloud Security Posture Vulnerabilities Page - Grouping Default Grouping groups vulnerabilities by cloud account and sort by number of vulnerabilities desc
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eeaa-6bd8-4e82-b89f-74bf8562f849) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301104/jobs/0196eeaa-6bd8-4e82-b89f-74bf8562f849/artifacts/0196eee5-2be7-4e33-b048-8e8e1760fc3e) FTR Configs #<span></span>79 / console app misc console behavior keyboard shortcuts open documentation should open documentation when Ctrl+/ is pressed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eee5-6585-4613-97fe-40824d19fec7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301104/jobs/0196eee5-6585-4613-97fe-40824d19fec7/artifacts/0196eef0-79db-4a59-90f4-694fe02211b3) FTR Configs #<span></span>79 / console app misc console behavior keyboard shortcuts open documentation should open documentation when Ctrl+/ is pressed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301104#0196eeaa-6bd9-488e-85bd-515004e25692) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301104/jobs/0196eeaa-6bd9-488e-85bd-515004e25692/artifacts/0196eece-0570-44c7-b301-d62bb56bafee) FTR Configs #<span></span>80 / Machine learning rule execution logic API @<span></span>ess @<span></span>serverless @<span></span>skipInServerlessMKI Machine Learning Detection Rule - Manual rule run with interval suppression duration alerts when run on a time range that the rule has not previously seen, and deduplicates if run there more than once

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301104","number":301104,"commit":"a7065b22d1f54b111609beee0d0bcc1a66182b9e"}],"number":301104}
buildkite-pr-comment-->

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301318)
* Commit: 717ebba7ba9a2b8210a70a27717b07b35da4d71a

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78e8-415a-abc0-87d2adede877)
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78e8-4fba-9171-25a6d43da89d)
* [Jest Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78e9-46c2-9fad-f9b0fae0a234)
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78e9-4726-9fd3-c335bc6a1e14)
* [Jest Tests #7](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ea-4649-9256-af249237557b)
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ea-484a-a62f-c5ccc244db1a)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ea-4fea-a265-011dbbb4e6fc)
* [Jest Tests #12](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78eb-4aca-9e16-e85f20ac05da)
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78eb-4d34-ba19-4f3a3e47eb99)
* [Jest Tests #15](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78eb-4ec3-94eb-be28d8648c45)
* [Jest Tests #16](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ec-43ff-a813-0e610b6c647a)
* [Jest Tests #18](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ec-44ec-adec-86a5aceb9619)
* [Jest Tests #19](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ec-4eed-8565-626809efd553)
* [Jest Tests #20](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ec-44b4-8268-60346d9ae298)
* [Jest Tests #21](https://buildkite.com/elastic/kibana-pull-request/builds/301318#0196f2d9-78ec-461c-a3da-5f284385b67a)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/e8494adf77d201bd9e4be8a6ef8b48142935a61e) | [after](https://github.com/elastic/kibana/commit/717ebba7ba9a2b8210a70a27717b07b35da4d71a) | diff |
| --- | --- | --- | --- |
| `@kbn/reporting-common` | 89 | 90 | +1 |
| `taskManager` | 67 | 71 | +4 |
| total |  |  | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/e8494adf77d201bd9e4be8a6ef8b48142935a61e) | [after](https://github.com/elastic/kibana/commit/717ebba7ba9a2b8210a70a27717b07b35da4d71a) | diff |
| --- | --- | --- | --- |
| `fleet` | 1.7MB | 1.7MB | +30.0B |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/e8494adf77d201bd9e4be8a6ef8b48142935a61e) | [after](https://github.com/elastic/kibana/commit/717ebba7ba9a2b8210a70a27717b07b35da4d71a) | diff |
| --- | --- | --- | --- |
| `taskManager` | 9 | 8 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/e8494adf77d201bd9e4be8a6ef8b48142935a61e) | [after](https://github.com/elastic/kibana/commit/717ebba7ba9a2b8210a70a27717b07b35da4d71a) | diff |
| --- | --- | --- | --- |
| `reporting` | 49.5KB | 49.5KB | +30.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/e8494adf77d201bd9e4be8a6ef8b48142935a61e) | [after](https://github.com/elastic/kibana/commit/717ebba7ba9a2b8210a70a27717b07b35da4d71a) | diff |
| --- | --- | --- | --- |
| `@kbn/reporting-common` | 99 | 100 | +1 |
| `taskManager` | 113 | 117 | +4 |
| total |  |  | +5 |


</details>


### History
* :broken_heart: [Build #301104](https://buildkite.com/elastic/kibana-pull-request/builds/301104) failed a7065b22d1f54b111609beee0d0bcc1a66182b9e

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301318","number":301318,"commit":"717ebba7ba9a2b8210a70a27717b07b35da4d71a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301104","number":301104,"commit":"a7065b22d1f54b111609beee0d0bcc1a66182b9e"}],"number":301318}
buildkite-pr-comment-->

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/302069)
* Commit: 37e7fd219e484772dbe9b0ce6b2a495c4f4531b9

### Failed CI Steps
* [FTR Configs #74](https://buildkite.com/elastic/kibana-pull-request/builds/302069#0196fd84-7056-4f22-9115-b853c6ad67f5)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/302069#0196fd84-7056-4f22-9115-b853c6ad67f5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/302069/jobs/0196fd84-7056-4f22-9115-b853c6ad67f5/artifacts/0196fda2-e00a-4bd2-b6e1-b882508d3746) FTR Configs #<span></span>74 / lens serverless - group 1 - subgroup 1 lens smokescreen tests should transition from a multi-layer stacked bar to a multi-layer line chart and correctly remove all layers

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/7e3bb0b6a1cdf7cce173ca5dafff22fb0d332d56) | [after](https://github.com/elastic/kibana/commit/37e7fd219e484772dbe9b0ce6b2a495c4f4531b9) | diff |
| --- | --- | --- | --- |
| `@kbn/reporting-common` | 89 | 90 | +1 |
| `taskManager` | 67 | 71 | +4 |
| total |  |  | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7e3bb0b6a1cdf7cce173ca5dafff22fb0d332d56) | [after](https://github.com/elastic/kibana/commit/37e7fd219e484772dbe9b0ce6b2a495c4f4531b9) | diff |
| --- | --- | --- | --- |
| `fleet` | 1.7MB | 1.7MB | +30.0B |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/7e3bb0b6a1cdf7cce173ca5dafff22fb0d332d56) | [after](https://github.com/elastic/kibana/commit/37e7fd219e484772dbe9b0ce6b2a495c4f4531b9) | diff |
| --- | --- | --- | --- |
| `taskManager` | 9 | 8 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/7e3bb0b6a1cdf7cce173ca5dafff22fb0d332d56) | [after](https://github.com/elastic/kibana/commit/37e7fd219e484772dbe9b0ce6b2a495c4f4531b9) | diff |
| --- | --- | --- | --- |
| `reporting` | 49.5KB | 49.5KB | +30.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/7e3bb0b6a1cdf7cce173ca5dafff22fb0d332d56) | [after](https://github.com/elastic/kibana/commit/37e7fd219e484772dbe9b0ce6b2a495c4f4531b9) | diff |
| --- | --- | --- | --- |
| `@kbn/reporting-common` | 99 | 100 | +1 |
| `taskManager` | 113 | 117 | +4 |
| total |  |  | +5 |


</details>


### History
* :yellow_heart: [Build #301318](https://buildkite.com/elastic/kibana-pull-request/builds/301318) was flaky 717ebba7ba9a2b8210a70a27717b07b35da4d71a
* :broken_heart: [Build #301104](https://buildkite.com/elastic/kibana-pull-request/builds/301104) failed a7065b22d1f54b111609beee0d0bcc1a66182b9e

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302069","number":302069,"commit":"37e7fd219e484772dbe9b0ce6b2a495c4f4531b9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301318","number":301318,"commit":"717ebba7ba9a2b8210a70a27717b07b35da4d71a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301104","number":301104,"commit":"a7065b22d1f54b111609beee0d0bcc1a66182b9e"}],"number":302069}
buildkite-pr-comment-->

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

ymao1 committed:
[Response Ops][Reporting] Scheduled Reports - Task (merging into feature branch) (#219770)

Towards https://github.com/elastic/kibana/issues/216313

> [!Note]
> This PR will be merged into a [feature
branch](https://github.com/elastic/kibana/pull/221028)

## Summary

This PR adds the `scheduled_report` task, which generates reports on a
recurring cadence.
* maxConcurrency of this task is 1 so only one can run at a time
* this task shares concurrency with the existing `report:execute` task,
so only one of either task type can run at a time
* this task requires an API key to run so it will only run when ES
security and API keys are enabled and a permanent encryption key for
Kibana is set
* when the task runs, it creates a new `ReportSource` document in the
`.kibana-reporting` index that stores the final base64 encoded generated
output. This `ReportSource` matches the data model of existing reports
with the addition of a `scheduled_report_id` field which links to the
scheduled report saved object ID.

## Verify

* Use the dev console to schedule some reports.
* Verify the the reports get generated at the expected cadence
* Verify that the correct time range is used for each scheduled report

Note, the UI has not been updated to differentiate between scheduled and
single run reports. They will look the same from in the report listing,
except the scheduled report will have a timestamp appended to the report
name.

In the cloud deployment for this PR, we have an example scheduled report
that is running every day at 4:45 EDT:
https://kibana-pr-219770.kb.us-west2.gcp.elastic-cloud.com/app/management/insightsAndAlerting/reporting

---------

Co-authored-by: Ersin Erdal <ersin.erdal@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ersin Erdal <92688503+ersin-erdal@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

ymao1 committed:
Merge

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306272)
* Commit: 12d806411d33fe4153b387c498a7264a75f2e2c5
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #303860](https://buildkite.com/elastic/kibana-pull-request/builds/303860) was flaky 6010d6a0b9603fa693d535e15207a99c3cb71151
* :yellow_heart: [Build #303478](https://buildkite.com/elastic/kibana-pull-request/builds/303478) was flaky 577b51739fd1ec0df3de4997d225012a7585fdab
* :yellow_heart: [Build #302069](https://buildkite.com/elastic/kibana-pull-request/builds/302069) was flaky 37e7fd219e484772dbe9b0ce6b2a495c4f4531b9
* :yellow_heart: [Build #301318](https://buildkite.com/elastic/kibana-pull-request/builds/301318) was flaky 717ebba7ba9a2b8210a70a27717b07b35da4d71a
* :broken_heart: [Build #301104](https://buildkite.com/elastic/kibana-pull-request/builds/301104) failed a7065b22d1f54b111609beee0d0bcc1a66182b9e

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306272","number":306272,"commit":"12d806411d33fe4153b387c498a7264a75f2e2c5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303860","number":303860,"commit":"6010d6a0b9603fa693d535e15207a99c3cb71151"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303478","number":303478,"commit":"577b51739fd1ec0df3de4997d225012a7585fdab"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302069","number":302069,"commit":"37e7fd219e484772dbe9b0ce6b2a495c4f4531b9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301318","number":301318,"commit":"717ebba7ba9a2b8210a70a27717b07b35da4d71a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301104","number":301104,"commit":"a7065b22d1f54b111609beee0d0bcc1a66182b9e"}],"number":306272}
buildkite-pr-comment-->

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306953)
* Commit: eb2cb214790a4d83fb1812e0da193aac7bf90e65
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #306272](https://buildkite.com/elastic/kibana-pull-request/builds/306272) succeeded 12d806411d33fe4153b387c498a7264a75f2e2c5
* :yellow_heart: [Build #303860](https://buildkite.com/elastic/kibana-pull-request/builds/303860) was flaky 6010d6a0b9603fa693d535e15207a99c3cb71151
* :yellow_heart: [Build #303478](https://buildkite.com/elastic/kibana-pull-request/builds/303478) was flaky 577b51739fd1ec0df3de4997d225012a7585fdab
* :yellow_heart: [Build #302069](https://buildkite.com/elastic/kibana-pull-request/builds/302069) was flaky 37e7fd219e484772dbe9b0ce6b2a495c4f4531b9
* :yellow_heart: [Build #301318](https://buildkite.com/elastic/kibana-pull-request/builds/301318) was flaky 717ebba7ba9a2b8210a70a27717b07b35da4d71a

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306953","number":306953,"commit":"eb2cb214790a4d83fb1812e0da193aac7bf90e65"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306272","number":306272,"commit":"12d806411d33fe4153b387c498a7264a75f2e2c5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303860","number":303860,"commit":"6010d6a0b9603fa693d535e15207a99c3cb71151"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303478","number":303478,"commit":"577b51739fd1ec0df3de4997d225012a7585fdab"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302069","number":302069,"commit":"37e7fd219e484772dbe9b0ce6b2a495c4f4531b9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301318","number":301318,"commit":"717ebba7ba9a2b8210a70a27717b07b35da4d71a"}],"number":306953}
buildkite-pr-comment-->

ymao1 committed:
[Response Ops][Reporting] Scheduled Reports - List and Disable API (#220922)

Towards https://github.com/elastic/kibana/issues/216313

> [!Note]
> This PR will be merged into a [feature
branch](https://github.com/elastic/kibana/pull/221028)

## Summary

Adds internal APIs for listing and disabling scheduled reports. For this
initial iteration, we are not adding an `enable` API.

This adds a `Manage Scheduled Reports` feature privilege that, if the
user has it, allows them to view and disable the scheduled reports of
any user in the current space.

## To Verify

1. Create a user (`user1`) with the ability to generate reports but not
manage reports. Use the schedule API to schedule some reports as this
user.
2. Create another user (`user2`) with the ability to manage reports. Use
the schedule API to schedule some reports as this user.
3. As `user1`, use the Dev Console to list scheduled reports. You should
only see those created by `user1`. `user1` should also be able to
disable their own reports.
4. As `user2`, use the Dev Console to list scheduled reports. You should
see reports from both `user1` and `user2`. `user2` should be able to
disable a report created by `user1`.

---------

Co-authored-by: Ersin Erdal <ersin.erdal@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ersin Erdal <92688503+ersin-erdal@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307149)
* Commit: c6e35e773ef5ae0957184d3058b74a7a86d2e0f6
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #306953](https://buildkite.com/elastic/kibana-pull-request/builds/306953) succeeded eb2cb214790a4d83fb1812e0da193aac7bf90e65
* :green_heart: [Build #306272](https://buildkite.com/elastic/kibana-pull-request/builds/306272) succeeded 12d806411d33fe4153b387c498a7264a75f2e2c5
* :yellow_heart: [Build #303860](https://buildkite.com/elastic/kibana-pull-request/builds/303860) was flaky 6010d6a0b9603fa693d535e15207a99c3cb71151
* :yellow_heart: [Build #303478](https://buildkite.com/elastic/kibana-pull-request/builds/303478) was flaky 577b51739fd1ec0df3de4997d225012a7585fdab
* :yellow_heart: [Build #302069](https://buildkite.com/elastic/kibana-pull-request/builds/302069) was flaky 37e7fd219e484772dbe9b0ce6b2a495c4f4531b9

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307149","number":307149,"commit":"c6e35e773ef5ae0957184d3058b74a7a86d2e0f6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306953","number":306953,"commit":"eb2cb214790a4d83fb1812e0da193aac7bf90e65"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306272","number":306272,"commit":"12d806411d33fe4153b387c498a7264a75f2e2c5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303860","number":303860,"commit":"6010d6a0b9603fa693d535e15207a99c3cb71151"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303478","number":303478,"commit":"577b51739fd1ec0df3de4997d225012a7585fdab"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/302069","number":302069,"commit":"37e7fd219e484772dbe9b0ce6b2a495c4f4531b9"}],"number":307149}
buildkite-pr-comment-->

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

ymao1 committed:
Checking licensing for scheduled reports feature

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308092)
* Commit: f5f9d9daedcd18447b6a02335a53631a44413788
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #307149](https://buildkite.com/elastic/kibana-pull-request/builds/307149) succeeded c6e35e773ef5ae0957184d3058b74a7a86d2e0f6
* :green_heart: [Build #306953](https://buildkite.com/elastic/kibana-pull-request/builds/306953) succeeded eb2cb214790a4d83fb1812e0da193aac7bf90e65
* :green_heart: [Build #306272](https://buildkite.com/elastic/kibana-pull-request/builds/306272) succeeded 12d806411d33fe4153b387c498a7264a75f2e2c5
* :yellow_heart: [Build #303860](https://buildkite.com/elastic/kibana-pull-request/builds/303860) was flaky 6010d6a0b9603fa693d535e15207a99c3cb71151
* :yellow_heart: [Build #303478](https://buildkite.com/elastic/kibana-pull-request/builds/303478) was flaky 577b51739fd1ec0df3de4997d225012a7585fdab
* :yellow_heart: [Build #302069](https://buildkite.com/elastic/kibana-pull-request/builds/302069) was flaky 37e7fd219e484772dbe9b0ce6b2a495c4f4531b9

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308092","number":308092,"commit":"f5f9d9daedcd18447b6a02335a53631a44413788"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307149","number":307149,"commit":"c6e35e773ef5ae0957184d3058b74a7a86d2e0f6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306953","number":306953,"commit":"eb2cb214790a4d83fb1812e0da193aac7bf90e65"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306272","number":306272,"commit":"12d806411d33fe4153b387c498a7264a75f2e2c5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303860","number":303860,"commit":"6010d6a0b9603fa693d535e15207a99c3cb71151"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303478","number":303478,"commit":"577b51739fd1ec0df3de4997d225012a7585fdab"}],"number":308092}
buildkite-pr-comment-->

ymao1 committed:
[Response Ops][Reporting] Scheduled Reports - Audit Logging (merging into feature branch) (#221846)

Towards https://github.com/elastic/kibana/issues/216313

> [!Note]
> This PR will be merged into a [feature
branch](https://github.com/elastic/kibana/pull/221028)

## Summary

Logs audit events when users use the schedule/list/disable scheduled
reports APIs

## To Verify

1. Set `xpack.security.audit.enabled: true` in Kibana config and start
ES and Kibana
2. Use the API to schedule some reports and then list and disable using
the API

For example:

```
POST kbn:/internal/reporting/schedule/printablePdfV2
{
    "schedule": {
        "rrule": {
            "freq": 3,
            "interval": 1,
            "byhour": [0,1,2],
            "byminute": [50,55]
        }
    },
    "jobParams":"(browserTimezone:America/New_York,layout:(dimensions:(height:2220,width:1409),id:preserve_layout),locatorParams:!((id:DASHBOARD_APP_LOCATOR,params:(dashboardId:edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b,preserveSavedFilters:!t,timeRange:(from:now-7d/d,to:now),useHash:!f,viewMode:view))),objectType:dashboard,title:'[Logs] Web Traffic',version:'9.1.0')"
}

GET kbn:/internal/reporting/scheduled/list

PATCH kbn:/internal/reporting/scheduled/bulk_disable
{
    "ids": ["<reportId>"]
}
```

3. In your kibana folder, go to `logs/audit.log`. You should be able to
see some audit events logged, for example

```
{"event":{"action":"scheduled_report_schedule","category":["database"],"type":["creation"],"outcome":"unknown"},"kibana":{"space_id":"default","session_id":"m82PsJ/ryDwhpCtaoQQkFSjEthWTlMshadIeMY5l+Ww=","saved_object":{"type":"scheduled_report","id":"c871e247-5dd4-4f0a-8419-e70eb184645a","name":"PDF"}},"user":{"id":"u_mGBROF_q5bmFCATbLXAcCwKa0k8JvONAwSruelyKA5E_0","name":"elastic","roles":["superuser"]},"trace":{"id":"e255102c-6ef9-4264-9cbd-9333a0910577"},"client":{"ip":"127.0.0.1"},"service":{"node":{"roles":["background_tasks","ui"]}},"ecs":{"version":"8.11.0"},"@timestamp":"2025-06-10T15:06:55.369-04:00","message":"User is creating scheduled report [id=c871e247-5dd4-4f0a-8419-e70eb184645a] [name=PDF]","log":{"level":"INFO","logger":"plugins.security.audit.ecs"},"process":{"pid":95125,"uptime":359.058969666},"transaction":{"id":"1fd4a03b23a4ad0d"}}
{"event":{"action":"scheduled_report_list","category":["database"],"type":["access"],"outcome":"success"},"kibana":{"space_id":"default","session_id":"m82PsJ/ryDwhpCtaoQQkFSjEthWTlMshadIeMY5l+Ww=","saved_object":{"type":"scheduled_report","id":"5f878787-1ea3-4704-b43c-b25e27376e5a"}},"user":{"id":"u_mGBROF_q5bmFCATbLXAcCwKa0k8JvONAwSruelyKA5E_0","name":"elastic","roles":["superuser"]},"trace":{"id":"7fc9d281-ea86-4d1b-b9f2-8f2143d058e0"},"client":{"ip":"127.0.0.1"},"service":{"node":{"roles":["background_tasks","ui"]}},"ecs":{"version":"8.11.0"},"@timestamp":"2025-06-10T15:13:03.960-04:00","message":"User has accessed scheduled report [id=5f878787-1ea3-4704-b43c-b25e27376e5a]","log":{"level":"INFO","logger":"plugins.security.audit.ecs"},"process":{"pid":95125,"uptime":727.648898875},"transaction":{"id":"bb9eabc53740c18c"}}
```

---------

Co-authored-by: Ersin Erdal <ersin.erdal@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ersin Erdal <92688503+ersin-erdal@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308589)
* Commit: 41883e1cb77bc403e5de1dabd18ea48ff7079d42
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #308092](https://buildkite.com/elastic/kibana-pull-request/builds/308092) was flaky f5f9d9daedcd18447b6a02335a53631a44413788
* :green_heart: [Build #307149](https://buildkite.com/elastic/kibana-pull-request/builds/307149) succeeded c6e35e773ef5ae0957184d3058b74a7a86d2e0f6
* :green_heart: [Build #306953](https://buildkite.com/elastic/kibana-pull-request/builds/306953) succeeded eb2cb214790a4d83fb1812e0da193aac7bf90e65
* :green_heart: [Build #306272](https://buildkite.com/elastic/kibana-pull-request/builds/306272) succeeded 12d806411d33fe4153b387c498a7264a75f2e2c5
* :yellow_heart: [Build #303860](https://buildkite.com/elastic/kibana-pull-request/builds/303860) was flaky 6010d6a0b9603fa693d535e15207a99c3cb71151

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308589","number":308589,"commit":"41883e1cb77bc403e5de1dabd18ea48ff7079d42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308092","number":308092,"commit":"f5f9d9daedcd18447b6a02335a53631a44413788"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307149","number":307149,"commit":"c6e35e773ef5ae0957184d3058b74a7a86d2e0f6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306953","number":306953,"commit":"eb2cb214790a4d83fb1812e0da193aac7bf90e65"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306272","number":306272,"commit":"12d806411d33fe4153b387c498a7264a75f2e2c5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303860","number":303860,"commit":"6010d6a0b9603fa693d535e15207a99c3cb71151"}],"number":308589}
buildkite-pr-comment-->

doakalexi committed:
[ResponseOps][Reporting] Send email with report attachment for scheduled reporting task (merging into feature branch) (#220539)

Resolves https://github.com/elastic/kibana/issues/216316

> [!Note]
> This PR will be merged into a [feature
branch](https://github.com/elastic/kibana/pull/221028)


## Summary

This PR updates the schedule reporting to flow to notify users once the
report is generated successfully and adds support for attachments in the
email connector. There is a 10 MB limit for schedule report attachments,
if the attachment is larger than 10MB it will throw an error and save it
in the scheduled report.

The flow for sending attachments using Nodemailer is simple, we just
append them to the request.
For Exchange, the flow is more complicated:
1. Create a draft email
2. If the size of the attachment is 3MB or less we can just upload the
base64 attachment to the draft email.
3. If the size of the attachment is > 3MB, we need to create an upload
session and convert the attachment to bytes and upload it in 2MB chunks.
After all the chunks have been uploaded we can close the upload session.
4. Send the draft email.


### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios


### To verify

1. Add a preconfigured connector kibana.yml and set
`notifications.connectors.default.email`. We need to test this for
Exchange and Nodemailer. Slack me for the config and secrets to test
Exchange. For nodemailer you can use maildev.
5. Start kibana and add sample data.
6. Create a scheduled report for the sample data dashboard in dev tools
7. Verify that the scheduled report is emailed as an attachment and
opens correctly.
8. Increase of the sample data dashboard to > 10MB and verify that there
is an error and the scheduled report is not emailed.

---------

Co-authored-by: Ersin Erdal <ersin.erdal@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ersin Erdal <92688503+ersin-erdal@users.noreply.github.com>
Co-authored-by: Ying <ying.mao@elastic.co>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309352)
* Commit: 0b0e3a5c9b51c390946803b2025edf9ecc63bf2a
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #308589](https://buildkite.com/elastic/kibana-pull-request/builds/308589) succeeded 41883e1cb77bc403e5de1dabd18ea48ff7079d42
* :yellow_heart: [Build #308092](https://buildkite.com/elastic/kibana-pull-request/builds/308092) was flaky f5f9d9daedcd18447b6a02335a53631a44413788
* :green_heart: [Build #307149](https://buildkite.com/elastic/kibana-pull-request/builds/307149) succeeded c6e35e773ef5ae0957184d3058b74a7a86d2e0f6
* :green_heart: [Build #306953](https://buildkite.com/elastic/kibana-pull-request/builds/306953) succeeded eb2cb214790a4d83fb1812e0da193aac7bf90e65
* :green_heart: [Build #306272](https://buildkite.com/elastic/kibana-pull-request/builds/306272) succeeded 12d806411d33fe4153b387c498a7264a75f2e2c5

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309352","number":309352,"commit":"0b0e3a5c9b51c390946803b2025edf9ecc63bf2a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308589","number":308589,"commit":"41883e1cb77bc403e5de1dabd18ea48ff7079d42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308092","number":308092,"commit":"f5f9d9daedcd18447b6a02335a53631a44413788"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307149","number":307149,"commit":"c6e35e773ef5ae0957184d3058b74a7a86d2e0f6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306953","number":306953,"commit":"eb2cb214790a4d83fb1812e0da193aac7bf90e65"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306272","number":306272,"commit":"12d806411d33fe4153b387c498a7264a75f2e2c5"}],"number":309352}
buildkite-pr-comment-->

nickpeihl reviewed (approved):
Canvas feature changes lgtm.

In testing, I found a [bug with Canvas PDFs](https://github.com/elastic/kibana/issues/224275) but that bug exists in main, so unrelated to this PR.

ymao1 committed:
[Response Ops][Reports] Updating list scheduled report API response (#224262)

> [!Note]
> This PR will be merged into a [feature
branch](https://github.com/elastic/kibana/pull/221028)

## Summary

Updating the list scheduled report API response to return `payload` and
`space_id`. This allows generating links for the reporting UI.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309533)
* Commit: 2d6e8be553e7e5762c64cbac0c4de354072813fb
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #309352](https://buildkite.com/elastic/kibana-pull-request/builds/309352) succeeded 0b0e3a5c9b51c390946803b2025edf9ecc63bf2a
* :green_heart: [Build #308589](https://buildkite.com/elastic/kibana-pull-request/builds/308589) succeeded 41883e1cb77bc403e5de1dabd18ea48ff7079d42
* :yellow_heart: [Build #308092](https://buildkite.com/elastic/kibana-pull-request/builds/308092) was flaky f5f9d9daedcd18447b6a02335a53631a44413788
* :green_heart: [Build #307149](https://buildkite.com/elastic/kibana-pull-request/builds/307149) succeeded c6e35e773ef5ae0957184d3058b74a7a86d2e0f6
* :green_heart: [Build #306953](https://buildkite.com/elastic/kibana-pull-request/builds/306953) succeeded eb2cb214790a4d83fb1812e0da193aac7bf90e65

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309533","number":309533,"commit":"2d6e8be553e7e5762c64cbac0c4de354072813fb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309352","number":309352,"commit":"0b0e3a5c9b51c390946803b2025edf9ecc63bf2a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308589","number":308589,"commit":"41883e1cb77bc403e5de1dabd18ea48ff7079d42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308092","number":308092,"commit":"f5f9d9daedcd18447b6a02335a53631a44413788"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307149","number":307149,"commit":"c6e35e773ef5ae0957184d3058b74a7a86d2e0f6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306953","number":306953,"commit":"eb2cb214790a4d83fb1812e0da193aac7bf90e65"}],"number":309533}
buildkite-pr-comment-->

doakalexi committed:
[ResponseOps] [Reporting] Update the email subject, message and attachment filename (merging into feature branch) (#224312)

## Summary

This PR updates the scheduled report email content.

Subject: `${report.title}-${timestamp} scheduled report`
Message: `Your scheduled report is attached for you to download or
share.`
Attachment filename: `${report.title}-${timestamp}.${extension}`

---------

Co-authored-by: Ersin Erdal <ersin.erdal@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ersin Erdal <92688503+ersin-erdal@users.noreply.github.com>
Co-authored-by: Ying <ying.mao@elastic.co>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309649)
* Commit: 3e78c243ff041098f7287013b96460527f0b641c
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #309533](https://buildkite.com/elastic/kibana-pull-request/builds/309533) succeeded 2d6e8be553e7e5762c64cbac0c4de354072813fb
* :green_heart: [Build #309352](https://buildkite.com/elastic/kibana-pull-request/builds/309352) succeeded 0b0e3a5c9b51c390946803b2025edf9ecc63bf2a
* :green_heart: [Build #308589](https://buildkite.com/elastic/kibana-pull-request/builds/308589) succeeded 41883e1cb77bc403e5de1dabd18ea48ff7079d42
* :yellow_heart: [Build #308092](https://buildkite.com/elastic/kibana-pull-request/builds/308092) was flaky f5f9d9daedcd18447b6a02335a53631a44413788
* :green_heart: [Build #307149](https://buildkite.com/elastic/kibana-pull-request/builds/307149) succeeded c6e35e773ef5ae0957184d3058b74a7a86d2e0f6

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309649","number":309649,"commit":"3e78c243ff041098f7287013b96460527f0b641c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309533","number":309533,"commit":"2d6e8be553e7e5762c64cbac0c4de354072813fb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309352","number":309352,"commit":"0b0e3a5c9b51c390946803b2025edf9ecc63bf2a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308589","number":308589,"commit":"41883e1cb77bc403e5de1dabd18ea48ff7079d42"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308092","number":308092,"commit":"f5f9d9daedcd18447b6a02335a53631a44413788"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307149","number":307149,"commit":"c6e35e773ef5ae0957184d3058b74a7a86d2e0f6"}],"number":309649}
buildkite-pr-comment-->

tsullivan commented on x-pack/platform/plugins/private/reporting/server/lib/tasks/run_scheduled_report.ts:
```diff
@@ -0,0 +1,200 @@
+
+  protected getMaxAttempts() {
+    return undefined;
+  }
```
How do max attempts for a scheduled report work?

tsullivan commented on x-pack/platform/plugins/private/reporting/server/lib/store/scheduled_report.test.ts:
```diff
@@ -0,0 +1,178 @@
+      kibanaId: 'instance-uuid',
+      kibanaName: 'kibana',
+      queueTimeout: 120000,
+      reportSO: {
```
I think it's a little confusing to mention `report saved object` because report documents aren't saved objects as they are commonly known in Kibana: they're just documents in a data stream. I see a lot of mentions in this PR like `reportSO` and in some error messages there are mentions of `report saved object`

tsullivan reviewed (commented):
Reviewed the code and left some comments. I haven't tried the new UX yet. I'm looking forward to it, this is very exciting!

ymao1 commented on x-pack/platform/plugins/private/reporting/server/lib/tasks/run_single_report.ts:
```diff
@@ -0,0 +1,162 @@
+          `Max attempts reached (${maxAttempts}). Queue timeout reached.`
+        );
+      }
+      await this.failJob(report, err);
```
sure, it can fail for the same reasons a "single" report job would fail

ymao1 commented:
> I haven't tried the new UX yet. 

@tsullivan Thanks for reviewing! The UX will be done in a separate PR. This PR just addresses the APIs and the task.

ymao1 committed:
PR feedback - improve readability

tsullivan reviewed (approved):
LGTM! Thank you so much!

I'm really looking forward to this new feature!

elena-shostak reviewed (approved):
LGTM!

Left a couple of nits

ymao1 committed:
PR feedback - name in audit log

ymao1 commented on x-pack/platform/plugins/private/reporting/server/routes/common/scheduled/scheduled_query.ts:
```diff
@@ -0,0 +1,356 @@
+          auditLogger.log(
+            scheduledReportAuditEvent({
+              action: ScheduledReportAuditAction.LIST,
+              savedObject: {
```
Updated in https://github.com/elastic/kibana/pull/221028/commits/48829a98649a1e0e5985dd5d0f343237adb65060. Thanks for catching! I had the name field in some audit events but not others for some reason 🤦‍♀️

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309913)
* Commit: 48829a98649a1e0e5985dd5d0f343237adb65060
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #309675](https://buildkite.com/elastic/kibana-pull-request/builds/309675) succeeded d8b80fde8db9f33999ecb8b124a066eb550b3bb5
* :green_heart: [Build #309649](https://buildkite.com/elastic/kibana-pull-request/builds/309649) succeeded 3e78c243ff041098f7287013b96460527f0b641c
* :green_heart: [Build #309533](https://buildkite.com/elastic/kibana-pull-request/builds/309533) succeeded 2d6e8be553e7e5762c64cbac0c4de354072813fb
* :green_heart: [Build #309352](https://buildkite.com/elastic/kibana-pull-request/builds/309352) succeeded 0b0e3a5c9b51c390946803b2025edf9ecc63bf2a
* :green_heart: [Build #308589](https://buildkite.com/elastic/kibana-pull-request/builds/308589) succeeded 41883e1cb77bc403e5de1dabd18ea48ff7079d42
* :yellow_heart: [Build #308092](https://buildkite.com/elastic/kibana-pull-request/builds/308092) was flaky f5f9d9daedcd18447b6a02335a53631a44413788

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309913","number":309913,"commit":"48829a98649a1e0e5985dd5d0f343237adb65060"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309675","number":309675,"commit":"d8b80fde8db9f33999ecb8b124a066eb550b3bb5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309649","number":309649,"commit":"3e78c243ff041098f7287013b96460527f0b641c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309533","number":309533,"commit":"2d6e8be553e7e5762c64cbac0c4de354072813fb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309352","number":309352,"commit":"0b0e3a5c9b51c390946803b2025edf9ecc63bf2a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308589","number":308589,"commit":"41883e1cb77bc403e5de1dabd18ea48ff7079d42"}],"number":309913}
buildkite-pr-comment-->

pmuellr commented:
I'm not seeing the notification bits work, assume I have something set up wrong.  I do have `notifications.connectors.default.email`, and seem to remember seeing it run at some point (outside of this PR work).  No messages or anything in the logs (running at DEBUG for reporting).

If someone else can validate a notification works, fine with me, I'll assume I've got some wonky setup.

pmuellr reviewed (approved):
LGTM, tested with multiple users with different management feature setting, and with the notify.  Quite nice!  Can't wait to get a UX on here!

js-jankisalvi reviewed (approved):
Verified locally, works as expected 👍

ymao1 committed:
Merge branch 'main' of github.com:elastic/kibana into scheduled-reports

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310398)
* Commit: 3db7db636bdac20b4a37a18d7226ad2e6646efeb
* [Cloud Deployment](https://kibana-pr-221028.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #309913](https://buildkite.com/elastic/kibana-pull-request/builds/309913) succeeded 48829a98649a1e0e5985dd5d0f343237adb65060
* :green_heart: [Build #309675](https://buildkite.com/elastic/kibana-pull-request/builds/309675) succeeded d8b80fde8db9f33999ecb8b124a066eb550b3bb5
* :green_heart: [Build #309649](https://buildkite.com/elastic/kibana-pull-request/builds/309649) succeeded 3e78c243ff041098f7287013b96460527f0b641c
* :green_heart: [Build #309533](https://buildkite.com/elastic/kibana-pull-request/builds/309533) succeeded 2d6e8be553e7e5762c64cbac0c4de354072813fb
* :green_heart: [Build #309352](https://buildkite.com/elastic/kibana-pull-request/builds/309352) succeeded 0b0e3a5c9b51c390946803b2025edf9ecc63bf2a
* :green_heart: [Build #308589](https://buildkite.com/elastic/kibana-pull-request/builds/308589) succeeded 41883e1cb77bc403e5de1dabd18ea48ff7079d42

cc @ymao1

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310398","number":310398,"commit":"3db7db636bdac20b4a37a18d7226ad2e6646efeb"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309913","number":309913,"commit":"48829a98649a1e0e5985dd5d0f343237adb65060"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309675","number":309675,"commit":"d8b80fde8db9f33999ecb8b124a066eb550b3bb5"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309649","number":309649,"commit":"3e78c243ff041098f7287013b96460527f0b641c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309533","number":309533,"commit":"2d6e8be553e7e5762c64cbac0c4de354072813fb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309352","number":309352,"commit":"0b0e3a5c9b51c390946803b2025edf9ecc63bf2a"}],"number":310398}
buildkite-pr-comment-->

ymao1 commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/224692">](https://github.com/elastic/kibana/pull/224692)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.1 -->

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
cc: @ymao1
